hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -3036,22 +3036,67 @@ func deleteNamespace(ctx context.Context, c client.Client, ns *corev1.Namespace)
 	return true, nil
 }
 
-func deleteControlPlaneOperatorRBAC(ctx context.Context, c client.Client, rbacNamespace string, controlPlaneNamespace string) error {
-	rbacMeta := metav1.ObjectMeta{
-		Name:      "control-plane-operator-" + controlPlaneNamespace,
-		Namespace: rbacNamespace,
+func deleteControlPlaneOperatorRole(ctx context.Context, c client.Client, rbacNamespace string, controlPlaneNamespace string) error {
+	role := &rbacv1.Role{
```

Comment:
This is an awful lot of code to delete a single resource, should we maybe move this into a helper that just takes a `client.Object`?

Reply:
can we stop the bleeding first, then refactor?